Fixes #14714 - Make inherited Puppet Classes more apparent

bastilian opened the pull request:
To better recognise which Puppet Classes are inherited for Host
or Hostgroup they are under a sub-heading as a separated list and mentions the Hostgroup inherited from by name in the tooltip.

![new host group - puppet classes](https://cloud.githubusercontent.com/assets/7757/21608671/4e0a5e4e-d1be-11e6-8187-e1b7c600012c.png)

The issue suggests to have the Hostgroup the class is inherited from as part of the module name, should this be as following the pattern of `Parent-Hostgroup/module-name` or is the approach with with the header and the mention in the tooltip already a good approach to address #14714?

orrabin commented:
Maybe the Included classes title should be bigger? because when all the classes are inherited it looks like this:
![image](https://cloud.githubusercontent.com/assets/7894647/21612200/dd82fa6c-d1d8-11e6-9eba-745de5b2a857.png)

Do we still need the tooltip if the classes are now grouped under the title that explains where they came from?

bastilian commented:
@orrabin I've used a heading smaller than "Included Classes" on first instinct, since they are included, but inherited. Using a the same font size does make sense as well. In both cases though I think it would be nice to have a message that are no classes are added yet that are included, as it is when no environment is selected.

The tooltip makes sense, in my opinion, though I can't speak from experience (yet), when the list of modules grows and the heading is out of sight.

Rohoover commented:
Would it be worth making the Inherited Classes an accordion? Then also add field level help (http://www.patternfly.org/pattern-library/forms-and-controls/field-level-help/#_) to accommodate your tooltip? 

Also - I'm trying to get feedback now whether that blue icon is correct, or if we should use pficon_info.

bastilian commented:
@Rohoover We can try using an accordion for this, then it would make sense to do the include Classes as well, which is expanded by default. Would the field help then be on the title or with each puppet class?

Rohoover commented:
@bastilian If the tooltip content is the same for all puppet classes then I would put it on the title.

bastilian commented:
@Rohoover I will remove the tooltips. On the title it would be redundant, as it would say the same. The separation by the title should be enough, or?

Rohoover commented:
@bastilian I think the title should be enough

bastilian commented:
@Rohoover Did you mean the accordion to be used like this for the inherited classes:

![screen shot 2017-01-09 at 09 25 32](https://cloud.githubusercontent.com/assets/7757/21760626/c3f7a830-d64d-11e6-89e0-395fe4dd9f1e.png)

Rohoover commented:
@bastilian 

No, just with the chevron/carrot.

dLobatog commented:
@bastilian Setting to WoC to change the accordion by the chevron/carrot + fixing the broken test

dLobatog commented:
@Rohoover Shouldn't it be h3 like the other headers as @orrabin suggested, even with the chevron? It looks a bit odd otherwise IMO

![screenshot from 2017-01-11 13-00-18](https://cloud.githubusercontent.com/assets/598891/21847790/13677e3e-d7fe-11e6-95bf-17cfa857e6bd.png)

Rohoover commented:
@dLobatog 
Yes, it looks like it should be the same as "Included Classes" and "Available Classes"

bastilian commented:
@Rohoover but with the chevron/carrot to expand and collapse?

Rohoover commented:
@bastilian 
Yes, unless you aren't concerned with how long the list would be.

bastilian commented:
@Rohoover I think @dLobatog can make better judgement of that.

In either case putting it into an accordion would be useful, then the question is only to collapse it or not by default.

Rohoover commented:
@bastilian 
I would collapse by default.

bastilian commented:
@Rohoover The inherited classes are now in an accordion, collapsed by default and styled to fit in better:
![screen shot 2017-01-11 at 17 40 17](https://cloud.githubusercontent.com/assets/7757/21863047/20bceeaa-d83c-11e6-8acd-a0e3b5a79d7a.png)


Even thought the title styles are now aligned it might still be a good idea to have a "placeholder" for included classes:

![screen shot 2017-01-11 at 17 40 32](https://cloud.githubusercontent.com/assets/7757/21863049/20d9a784-d83c-11e6-9ab8-353519e71c3f.png)

With classes it looks good:

![screen shot 2017-01-11 at 17 40 55](https://cloud.githubusercontent.com/assets/7757/21863050/20eac58c-d83c-11e6-945d-2efbffb24f7f.png)

The styles to remove borders and backgrounds can be applied to any accordion via the `.paneless` class and are separate from the title specific styling.

Rohoover commented:
@bastilian 

![bitmoji](https://render.bitstrips.com/v2/cpanel/10215973-131321122_2-s4-v1.png?transparent=1&palette=1&width=246)

dLobatog commented:
@bastilian Please check out integration tests failures - http://ci.theforeman.org/job/test_develop_pr_core/11740/ - should be ready to merge afterwards. Thanks!

bastilian committed:
Fixes #14714 - Make inherited Puppet Classes more apparent

To better recognize which Puppet Classes are inherited for Host
or Hostgroup they are under a sub-heading as a seperated list.

shlomizadok commented:
@Rohoover where do you get those cool images from?

Rohoover commented:
@shlomizadok 

It's a bitmoji, and I have the chrome plugin that adds a button to tool bars so I can insert them.
<img width="659" alt="screen shot 2017-01-12 at 9 33 46 am" src="https://cloud.githubusercontent.com/assets/18266453/21893707/3f81f36c-d8aa-11e6-969e-02f4807449c4.png">

dLobatog commented:
Thanks @bastilian , @orrabin & @Rohoover !
